### PR TITLE
fixed download_sample_data

### PR DESCRIPTION
**Pull request by gcetusic**
Importing sunpy.data.sample notifies that no sample data is available and should be downloaded with download_sample_data but once you try that is fails when calling rename with 'invalid cross-device link' when using rename, probably because of different partitions of files. Also, the readme doesn't work because sample image that gets called by matplolib has moved to sunpy.data.sample and minor PEP8 fixes.

### CODE REVIEW

File: sunpy/data/_sample.py [hunk 1]
```diff
@@ -3,10 +3,11 @@
...
 
 from sunpy.util.net import url_exists
-from os import rename, remove
+from os import remove
 from astropy.utils.data import download_file
 from zipfile import ZipFile
 from urllib2 import URLError
+from shutil import move
 
 import os.path
```
Comment (dpshelio) [inline sunpy/data/_sample.py]: Not to do particularly with what's been changed in this PR, but following PEP:8ball: should the order of the imports be rearranged here? Like:
`__future__, os, shutil, zipfile, urllib2`  first
`astropy` then 
and `sunpy` aftewards?

**Comment by ehsteve** [comment]
:+1: Thanks!

File: sunpy/data/_sample.py (hunk 1 above)
Comment (Cadair) [inline sunpy/data/_sample.py]: yes, the imports should be in a different order. 

```python
__future__

stdlib

external

sunpy
```

Comment (gcetusic) [inline sunpy/data/_sample.py]: I'll have to confirm that but I think you're right. Linter can't detect that.

**Comment by dpshelio** [comment]
:+1: Thanks for the imports reordering :)

**Comment by Cadair** [comment]
:+1:

**Pull request merged**